mart/datamodules/coco.py

```diff
@@ -89,3 +95,23 @@ def __getitem__(self, index: int):
 # Source: https://github.com/pytorch/vision/blob/dc07ac2add8285e16a716564867d0b4b953f6735/references/detection/utils.py#L203
 def collate_fn(batch):
     return tuple(zip(*batch))
+
+
+def yolo_collate_fn(batch):
+    images, targets = collate_fn(batch)
+
+    # Collate images
+    images = default_collate(images)
+
+    # Turn tuple of dicts into dict of tuples
+    new_targets = {k: tuple(t[k] for t in targets) for k in targets[0].keys()}
+    new_targets["list_of_targets"] = targets
```

Comment:
Would be nice to get rid of this...

Reply:
Basically, we either need to decide to accept a `List[Tensor]` for `input` and `List[Dict[str, Tensor]]` for `target` or `Tensor` for `input` and `Dict[str, Tensor]` for `target`. The former interface at least allows images with different shapes and sizes even if the underlying network does not.